x/concentrated-liquidity/swapstrategy/zero_for_one.go

```diff
@@ -171,24 +176,13 @@ func (s zeroForOneStrategy) InitializeNextTickIterator(ctx sdk.Context, poolId u
 			iter.Close()
 			panic(fmt.Errorf("invalid tick index (%s): %v", string(iter.Key()), err))
 		}
-		if tick < currentTickIndexPlusOne {
+		if tick <= currentTickIndex {
```

Comment:
Note to reviewer: read the PR description why it is correct to be inclusive of the current tick.

Currently, we have the same invariant hold. This is changed to remove the obsolete `InitializeTickValue` method since we can assure inclusivity here.

Reply:
Is there any case where this if statement wouldn't hold? (Is this just a sanity check?)

Reply:
I believe there is.

Imagine we create a first narrow position at the bottom of the tick range.

The current tick can be initialized to be above the position's bucket.

Next, assume zero for one swap (left). When we search for the upper tick of the position's bucket at the beginning of the swap, `currentTickIndex` can be strictly greater than `tck`